test/TestMatchError.py

```diff
@@ -0,0 +1,44 @@
+from ansiblelint.errors import MatchError
+from ansiblelint.rules import AnsibleLintRule
+from ansiblelint.rules.AlwaysRunRule import AlwaysRunRule
+from ansiblelint.rules.BecomeUserWithoutBecomeRule import BecomeUserWithoutBecomeRule
+
+import pytest
+
+
+def test_matcherror_compare():
```

Comment:
This needs to be parametrized with all of the args because it doesn't catch a bug with `self.line` missing from hash.

Reply:
Not really, see last change where I document that line dictionary! is not part of the hashing logic.

Reply:
Then it should be added to tests showing that it's expected.